server/src/test/java/org/apache/druid/segment/realtime/appenderator/StreamAppenderatorDriverFailTest.java

```diff
@@ -221,10 +221,8 @@ public void testFailDuringDrop() throws IOException, InterruptedException, Timeo
 
     Assert.assertNull(driver.startJob(null));
 
-    for (int i = 0; i < ROWS.size(); i++) {
-      committerSupplier.setMetadata(i + 1);
-      Assert.assertTrue(driver.add(ROWS.get(i), "dummy", committerSupplier, false, true).isOk());
-    }
+    committerSupplier.setMetadata(1);
```

Comment:
This change is intentional to avoid any uncertainty that may arise due to the order of segment ids.
By adding a single row, exactly one segment is created and we can verify its id in the exception message.